hack/flags.sh

```diff
@@ -60,4 +60,23 @@ parse_flags() {
     exit 1
   fi
   export ROOT_WORKSPACE=${ROOT_WORKSPACE:-"root"}
+
+  # Check config files and version compatibility
+  if kubectl version -o yaml --kubeconfig ${CLUSTER_KUBECONFIG} | yq '.serverVersion.gitVersion' | grep -q kcp; then
+    echo CLUSTER_KUBECONFIG=${CLUSTER_KUBECONFIG} points to KCP not to cluster.
+    exit 1
+  fi
+  KCP_SERVER_VERSION=$(kubectl version -o yaml --kubeconfig ${KCP_KUBECONFIG} | yq '.serverVersion.gitVersion')
+  if ! echo "$KCP_SERVER_VERSION" | grep -q kcp; then
+    echo KCP_KUBECONFIG=${KCP_KUBECONFIG} does not point to KCP cluster.
+    exit 1
+  fi
+  KCP_SERVER=$(echo $KCP_SERVER_VERSION | sed 's/.*kcp-v\(.*\)\..*/\1/')
+  KCP_CLIENT=$(kubectl kcp --version | sed 's/.*kcp-v\(.*\)\..*/\1/')
```

Comment:
Also this part should be skipped if the kcp is running locally, otherwise, it would fail as `v0.0.0-master+$Format:%H$` doesn't contain the `kcp` string

Reply:
added check also for kcp plugin ersion

Reply:
I didn't mean the KCP_CLIENT, I meant the whole part starting with line 70, because KCP_SERVER_VERSION is what matters here. As soon as KCP_SERVER_VERSION contains `v0.0.0-master`, then we should skip the check for the client versions.
Sorry for not being clear

Reply:
ok, updated